### PR TITLE
Assign multiple values as array instead of stream.

### PR DESCRIPTION
**Pull request by anu3990**
So we can incorporate your pull request, please share the following:
* What issue are you addressing with this pull request? https://project.marklogic.com/jira/browse/JAVA-62
* Are you modifying the correct branch? develop-failover
* Have you run unit tests? Yes
* Version of MarkLogic Java Client API (see Readme.txt)
* Version of MarkLogic Server 9.0.9
* Java version (`java -version`)
* OS and version
* What Changed: What happened before this change? What happens without this change?

### CODE REVIEW

File: marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java [hunk 1]
```diff
@@ -5759,17 +5760,22 @@ private RequestBody makeRequestBody(CallField[] params) {
...
         } else if (param instanceof SingleNodeCallField) {
-          AbstractWriteHandle paramValue = ((SingleNodeCallField) param).getParamValue();
+          AbstractWriteHandle paramValue = (AbstractWriteHandle) ((SingleNodeCallField) param).getParamValue();
           if (paramValue != null) {
             HandleImplementation handleBase = HandleAccessor.as(paramValue);
             if(! handleBase.isResendable()) {
-              hasStreamingPartCondition.set();
+                if(param instanceof BufferedSingleNodeCallField) {
+                    ((BufferedSingleNodeCallField) param).setParamValue(new BytesHandle((BufferableHandle) paramValue));
+                    handleBase = HandleAccessor.as(((SingleNodeCallField) param).getParamValue());
```
Comment (ehennum) [inline marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java]: Would this section require fewer casts and thus be easier to understand if a variable captured the BytesHandle?  As in:

     BytesHandle bytesHandle = new BytesHandle((BufferableHandle) handleBase);
     ((BufferedSingleNodeCallField) param).setParamValue(bytesHandle);
    handleBase = bytesHandle;

File: marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java [hunk 2]
```diff
@@ -5782,7 +5788,23 @@ private RequestBody makeRequestBody(CallField[] params) {
                   multiBldr.addFormDataPart(paramName, null, makeRequestBody(paramValue));
                 });
           }
-        } else {
+        } else if (param instanceof BufferedMultipleNodeCallField) {
+            BufferableHandle[] paramValues = ((BufferedMultipleNodeCallField) param).getParamValuesArray();
+            if(paramValues != null) {
```
Comment (ehennum) [inline marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java]: Instead of constructing a stream, could a loop perform the same operations just as easily:

    for (BufferableHandle paramValue: paramValues) {
        // continue if null
        // same code as body of the forEachOrdered
    }

File: marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/MultipleValuesAsArrayTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        BufferableHandle handle = null;
+        assertNull(NodeConverter.copyToBytesHandle(handle));
+
+        BufferableHandle[] handleArray = null;
+        assertNull(NodeConverter.copyToBytesHandle(handleArray));
+
+        handleArray = new BufferableHandle[1];
+        handleArray[0] = handle;
+        assertNotNull(NodeConverter.copyToBytesHandle(handleArray));
+        assertNull(NodeConverter.copyToBytesHandle(handleArray)[0]);
```
Comment (ehennum) [inline marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/MultipleValuesAsArrayTest.java]: Good to have this test of the utility method.

Would it be useful to add a test that a non-empty BufferableHandle is copied?  A StringHandle is an easy BufferableHandle to work with.

File: marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/MultipleValuesAsArrayTest.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        CallBatcher<Void, CallManager.ManyCallEvent<Reader>> batcher = manyCaller.batcher().forArgsGenerator(result -> {
+            if (result == null && output.counter == 0) {
+                output.counter++;
+                output.args = manyCaller.args().param("param1", handles);
+                return output.args;
+            } else if (output.counter == 1) {
+                output.counter++;
+                assertEquals(output.args, result.getArgs());
+                return result.getArgs();
+            } else {
```
Comment (ehennum) [inline marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/MultipleValuesAsArrayTest.java]: Great test.  Would it be useful to add an assertion that output.counter is 2.

File: marklogic-client-api/src/main/java/com/marklogic/client/impl/RESTServices.java [hunk 5]
```diff
@@ -563,28 +567,49 @@ public BufferedMultipleAtomicCallField(String paramName, Stream<String> paramVal
...
+	  private BufferableHandle[] paramValues;
+	  public BufferedMultipleNodeCallField(String paramName, Stream<BufferableHandle> paramValues) {
+	        this(paramName, paramValues.toArray(size -> new BufferableHandle[size]));
+	  }
+	  public BufferedMultipleNodeCallField(String paramName, BufferableHandle[] paramValues) {
+	      super(paramName);
+	      this.paramValues = paramValues;
 	  }
 	  @Override
 	  public Stream<BytesHandle> getParamValues() {
```
[the remark this answers is not in the document]
Comment (anu3990) [reply marklogic-client-api/src/main/java/com/marklogic/client/impl/RESTServices.java]: Hi Erik. Sorry i could not understand this one. Should I change the constructor BufferedMultipleNodeCallField(String paramName, BufferableHandle[] paramValues) to BufferedMultipleNodeCallField(String paramName, Stream<BufferableHandle> paramValues) or the datatype of paramValues from BufferableHandle[] to Stream<BufferableHandle> .?

[the remark this answers is not in the document]
Comment (ehennum) [reply marklogic-client-api/src/main/java/com/marklogic/client/impl/RESTServices.java]: Sorry, an editing error, I guess.

Let's change the signature of getParamValues() to Stream< BufferableHandle > so we can return Stream.streamOf(paramValues);
